src/osd/PrimaryLogPG.cc

```diff
@@ -15112,15 +15113,25 @@ int PrimaryLogPG::rep_repair_primary_object(const hobject_t& soid, OpContext *ct
   waiting_for_unreadable_object[soid].push_back(op);
   op->mark_delayed("waiting for missing object");
 
-  ceph_assert(is_clean());
-  state_set(PG_STATE_REPAIR);
-  state_clear(PG_STATE_CLEAN);
-  queue_peering_event(
-      PGPeeringEventRef(
-	std::make_shared<PGPeeringEvent>(
-	get_osdmap_epoch(),
-	get_osdmap_epoch(),
-	PeeringState::DoRecovery())));
+  if (is_clean()) {
+    dout(20) << __func__ << " First read error starting recovery for " << soid << dendl;
+    state_set(PG_STATE_REPAIR);
+    state_clear(PG_STATE_CLEAN);
+    queue_peering_event(
+        PGPeeringEventRef(
+	  std::make_shared<PGPeeringEvent>(
+	  get_osdmap_epoch(),
+	  get_osdmap_epoch(),
+	  PeeringState::DoRecovery())));
+  } else {
+    // Set repair in case we are the first read error and we happen to be
```

Comment:
Do you mean we are the first read error for the object `soid` and the PG is currently executing a recovery/backfill for a different object, let's say `soid_2` or for the same object `soid`?

Reply:
Recovery/backfill could be running because an osd came up or went down.  So multiple other objects could be getting processed.  It doesn't matter.  We are just adding an additional item that is a repair (setting flag for user info) to the task already running.